server/backend/cafeteriaRatingDBInitializer.go

```diff
@@ -0,0 +1,206 @@
+package backend
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/TUM-Dev/Campus-Backend/model"
+	"github.com/TUM-Dev/Campus-Backend/model/cafeteria_rating_models"
+	"github.com/guregu/null"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type MultiLanguageTags struct {
+	MultiLanguageTags []Tag `json:"tags"`
+}
+type Tag struct {
+	TagNameEnglish string `json:"tagNameEnglish"`
+	TagNameGerman  string `json:"tagNameGerman"`
+}
+
+type MultiLanguageNameTags struct {
+	MultiLanguageNameTags []NameTag `json:"tags"`
+}
+type NameTag struct {
+	TagNameEnglish string   `json:"tagNameEnglish"`
+	TagNameGerman  string   `json:"tagNameGerman"`
+	Notincluded    []string `json:"notincluded"`
+	Canbeincluded  []string `json:"canbeincluded"`
+}
+
+/*
+Writes all available tags from the json file into tables in order to make them easier to use.
+Will be executed once while the server is started.
+*/
+func initTagRatingOptions(db *gorm.DB) {
+	updateTagTable("backend/static_data/dishRatingTags.json", db, MEAL)
+	updateTagTable("backend/static_data/cafeteriaRatingTags.json", db, CAFETERIA)
+	updateNameTagOptions(db)
+	addEntriesForCronJob(db, "averageRatingComputation", 300)
+	addEntriesForCronJob(db, "dishNameDownload", 302400) //run twice every week
+}
+
+func addEntriesForCronJob(db *gorm.DB, cronName string, interval int32) {
+	var exists bool
+	res := db.Model(&model.Crontab{}).
+		Select("count(*) > 0").
+		Where("type LIKE ?", cronName).
+		Find(&exists).
+		Error
+
+	if res != nil {
+		log.Error(res.Error)
+	} else if !exists {
+
+		db.Model(&model.Crontab{}).
+			Create(&model.Crontab{
+				Interval: interval,
+				Type:     null.String{NullString: sql.NullString{String: cronName, Valid: true}},
+				LastRun:  0,
+			})
+	}
+}
+
+/*
+Updates the list of dishtags.
+If a Tag with the exact german and english name does not exist yet, it will be created.
+Old tags won't be removed to prevent problems with foreign keys.
+*/
+func updateNameTagOptions(db *gorm.DB) {
+	absPathDishNames, _ := filepath.Abs("backend/static_data/dishNameTags.json")
+	tagsNames := generateNameTagListFromFile(absPathDishNames)
+	var elementID int32
+	for _, v := range tagsNames.MultiLanguageNameTags {
+		var parentID int32
+
+		potentialTag := db.Model(&cafeteria_rating_models.DishNameTagOption{}).
+			Where("EN LIKE ? AND DE LIKE ?", v.TagNameEnglish, v.TagNameGerman).
+			Select("DishNameTagOption").
+			Scan(&parentID)
```

Comment:
Since the parentId is needed in the rest of the method. this scan can not be replaced by a count. However, the remaining scans in the method can be replaced.